src/codegen/enums.rs

```diff
@@ -150,6 +151,25 @@ fn generate_enum(env: &Env, w: &mut Write, enum_: &Enumeration, config: &GObject
 "
     ));
 
+    // Generate Display trait implementation.
+    try!(cfg_deprecated(w, env, enum_.deprecated_version, false, 0));
+    try!(version_condition(w, env, enum_.version, false, 0));
+    try!(writeln!(w,
+                  "impl fmt::Display for {0} {{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {{
+        write!(f, \"{0}::{{}}\", match *self {{", enum_.name));
+    for member in &members {
+        try!(cfg_deprecated(w, env, member.deprecated_version, false, 3));
+        try!(version_condition(w, env, member.version, false, 3));
+        try!(writeln!(w, "\t\t\t{0}::{1} => \"{1}\",", enum_.name, member.name));
```

Comment:
Two issues here
1) It should be configurable if this generated, just like what I added for other types. So that it can be manually overridden.
2) Use the "special functions" stuff from gir (just like we do for the various trait impls of other kinds of types). If there's e.g. a `to_string()` function, then this can be used here. I actually only thought (in the issue I created months ago) about creating the `Display` impl *only* if there's a special function for it.

Reply:
Note: currently we don't generate enums and flags functions, so this need be done before, if we decide generate only for some enums.

Reply:
That should happen because otherwise this is going to break the GStreamer bindings (where I have some custom `Display` impls) :)